src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -119,6 +120,12 @@ public Rotation2d getGyroscopeRotation() {
         return pigeon2.getRotation2d();
     }
 
+    public Rotation3d getGyroRotation(){
+        double ypr[] = {0,0,0};
+        pigeon2.getYawPitchRoll(ypr);
+        return new Rotation3d(ypr[2], ypr[1], ypr[0]);
```

Comment:
It looks like a Rotation3d requires units in [radians](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Rotation3d.html#%3Cinit%3E(double,double,double)), the Pigeon's [`getYawPitchRoll`](https://api.ctr-electronics.com/phoenix/release/java/com/ctre/phoenix/sensors/BasePigeon.html#getYawPitchRoll(double[])) method returns degrees. Be sure to convert units!

It also looks like the Rotation3d constructor's arguments are in the order Roll, Pitch, Yaw (the reverse of Pigeon's order)

Reply:
farhan sucks